src/Factory/ProductDocumentFactory.php

```diff
@@ -193,6 +183,51 @@ public function createFromSyliusSimpleProductModel(ProductInterface $syliusProdu
     }
 
     /**
+     * @param TaxonInterface $taxon
+     *
+     * @return TaxonDocument
+     */
+    private function createTaxonDocumentFromSyliusTaxon(TaxonInterface $taxon)
+    {
+        /** @var TaxonDocument $taxonDocument */
+        $taxonDocument = new $this->taxonDocumentClass();
+        $taxonDocument->setCode($taxon->getCode());
+        $taxonDocument->setSlug($taxon->getSlug());
```

Comment:
One more type we can have problem with translation.... We should take a TaxonTranslation for given locale first, and then gather translatable data